### ISSUE TITLE
Inconsistent matching results with XGBoost

### ISSUE THREAD

**Issue by riley-harper** [ipums/hlink#221]
We've seen some inconsistent results when running the same configuration file multiple times with the XGBoost classifier. We get mostly the same data, but with some minor variations and a different total count of predicted matches. After doing some investigating, I'm pretty sure that this is due to the fact that we're not setting the `seed` parameter to XGBoost, so its RNG gives different results each run. This makes the algorithm behave slightly differently on each run.

For reproducibility's sake, we should automatically set the `seed` parameter if the user does not set it in their config file. We already do this for the Spark machine learning models like RandomForest. We should start doing this for XGBoost and LightGBM as well.

The current implementation for the Spark models does not allow you to pass `seed` to models like RandomForest. If you do, it throws an error like this one.

```
An error occured:
ERROR type: <class 'TypeError'>
ERROR message: pyspark.ml.classification.RandomForestClassifier() got multiple values for keyword argument 'seed'
See log for details.
```

I think it'd be better to allow the user to pass in their own seed if they would like, but always set the seed to something by default. This makes the results consistent between runs but also lets users vary the seed if they would like to see how changing it affects the results.
